tests/fuzzer.c

```diff
@@ -572,16 +572,22 @@ static int FUZ_test(U32 seed, U32 nbCycles, const U32 startCycle, const double c
         FUZ_DISPLAYTEST;
         dict -= (FUZ_rand(&randState) & 0xF) + 1;   /* Separation, so it is an ExtDict */
         if (dict < (char*)CNBuffer) dict = (char*)CNBuffer;
+        /* we depend on exactly reproducing the compression operation,
```

Comment:
For my understanding : 
why would a difference in `resetStream()` produce a difference in compression result ?

Reply:
That's an interesting question. Here's the best theory I have (with @terrelln's help):

The fundamental difference is that when you start with an empty stream, hash table misses cause match checks against the beginning of the stream, while if you have a used stream--with a non-zero `currentOffset`--no match check occurs. If a hash miss points to the beginning of the stream and the bytes actually match, you will get a different output compared to a non-zero `currentOffset` stream where the hash miss doesn't trigger a match check.

This almost doesn't matter, since the starting position is hashed and written into the table at the beginning of the compression. However, because the hash is on 5 bytes and the initial match check is on 4 bytes, any position in which the first 4 bytes match the beginning of the stream but the fifth byte doesn't may hit a hash miss which hit this difference in behavior--the 0-offset stream will check against the beginning of the stream and find a match, which the non-0 stream won't.

Reply:
We can verify if this is the case by adding a loop like this to the beginning of the compression function and see if the test failure goes away. This will put position 0 on every value that could hash to it and actually match.

```c
if (ip == base) {
  BYTE bytes[5];
  memcpy(bytes, ip, 4); // before ip++
  for (U32 b = 0; b < 256; ++b) {
    bytes[4] = (BYTE)b;
    LZ4_putPosition(bytes, cctx->hashTable, tableType, bytes);
  }
}
```